lib/uklibid/Makefile.uk

```diff
@@ -38,4 +38,10 @@ LIBUKLIBID_SRCS-y += $(LIBUKLIBID_BASE)/exportsyms.awk>.uk
 LIBUKLIBID_EXPORTSYMS_AWKINCLUDES-y += $(LIBUKLIBID_BUILD)/libraries.in
 LIBUKLIBID_EXPORTS-y += $(LIBUKLIBID_BUILD)/exportsyms.uk
 
+EACHOLIB_SRCS-$(CONFIG_LIBUKLIBID_INFO) += $(LIBUKLIBID_BASE)/libinfo.S|libuklibid
```

Comment:
This will not be recompiled with if any other file of the library is altered. So the information embedded will only be correct for a clean build. Probably that is currently only relevant for the date/time?

```
$ make
  GEN     libuklibid: libraries.in.new
  CP      libuklibid: libraries.in
  CP      config
  CC      libukdebug: outf.o
  LD      libukdebug.ld.o
  OBJCOPY libukdebug.o
  LD      unikraft_qemu-arm64.dbg
  SCSTRIP unikraft_qemu-arm64
  UKBI    unikraft_qemu-arm64.bootinfo
  OBJCOPY unikraft_qemu-arm64.bin
  MKLINUX unikraft_qemu-arm64.img
  GZ      unikraft_qemu-arm64.gz
```

Reply:
Yeah, at the moment a weak point regarding DATE and TIME. We'd need to add another build system option to retrigger builds if something else in a lib changes. For fresh builds this is not an issue and PR #1119 is at least retriggering a rebuild on config changes (not code changes). I'd move this to a improvement at a later step later, because under certain circumstances the DATE and TIME information is correct if used (e.g., by a CI/CD doing a fresh build).

Reply:
Fair enough. It might be good to document this limitation, because it can be surprising for somebody just trying to use this feature.

Reply:
Nevermind, I just had the idea to handover the current time and date as compiler argument. I changed this in the PR. As soon as we properly detect changes of compile flags, the date and time should be always up to date.